File: Cpp/1491_Average_Salary_Excluding_the_Minimum_and_Maximum_Salary.cpp
class Solution {
public:
    double average(vector<int>& salary) {
        int max_salary = salary[0];
        int min_salary = salary[0];
        int sum = 0;
        for(auto& it : salary){
            max_salary = max(it, max_salary);
            min_salary = min(it, min_salary);
            sum += it;
        }
        
         return static_cast<double>(sum - max_salary - min_salary) / (salary.size()-2);
    }
};

File: Kotlin/1491_Average_Salary_Excluding_the_Minimum_and_Maximum_Salary.kt
class Solution {
    fun average(salary: IntArray): Double {
        var maxSalary = salary[0]
        var minSalary = salary[0]
        var sum = 0
        salary.forEach{
            maxSalary = maxOf(it, maxSalary)
            minSalary = minOf(it, minSalary)
            sum += it
        }
        
         return (sum.toDouble() - maxSalary - minSalary) / (salary.size - 2)
    }
}